File: wk6/py/validparentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        mp = {"}":"{", "]":"[",")":"("}

        for char in s:
            if char in "[({":
                stack.append(char)
                continue

            if not stack: return False

            if mp[char] == stack[-1]:
                stack.pop()
                continue
            
            return False

        return not stack


                

File: wk6/java/validParentheses.java
class Solution {
    public boolean isValid(String s) {
        LinkedList<Character> ll = new LinkedList<Character>();
        String open = "({["; String close = ")}]";
        for(int i = 0; i < s.length(); i++)
        {
            if(open.indexOf(s.charAt(i)) > -1)
                ll.push(s.charAt(i));
            else if(ll.peek() == null || open.indexOf(ll.pop()) != close.indexOf(s.charAt(i)))
                return false;
        }
        return ll.peek() == null;
    }
}